skins/oasis/modules/PageHeaderController.class.php

```diff
@@ -529,46 +509,45 @@ public function executeEditBox() {
 
 		// Editing: foo
 		$this->displaytitle = true;
-		$this->title = wfMsg( $msg, htmlspecialchars( $wgTitle->getPrefixedText() ) );
+		$this->title = wfMessage( $msg, $this->wg->Title->getPrefixedText() )->parse();
 
 		// back to article link
-		$this->subtitle = Wikia::link( $wgTitle, wfMsg( 'oasis-page-header-back-to-article' ), [ 'accesskey' => 'c' ], [ ], 'known' );
+		$this->subtitle = Linker::linkKnown( $this->wg->Title, wfMessage( 'oasis-page-header-back-to-article' )->escaped(), [ 'accesskey' => 'c' ] );
 	}
 
 	/**
-	 * Called instead of executeIndex when the CorporatePage extension is enabled.
+	 * Called instead of index when the CorporatePage extension is enabled.
 	 */
-	public function executeCorporate() {
-		global $wgTitle, $wgOut, $wgUser, $wgSuppressNamespacePrefix;
+	public function corporate() {
 		wfProfileIn( __METHOD__ );
 
-		$this->canAct = $wgUser->isAllowed( 'edit' );
+		$this->canAct = $this->wg->User->isAllowed( 'edit' );
 		if ( $this->canAct ) {
 			$this->prepareActionButton();
 			// dropdown actions
 			$this->dropdown = $this->getDropdownActions();
 		}
 
 		// page namespace
-		$ns = $wgTitle->getNamespace();
+		$ns = $this->wg->Title->getNamespace();
 
 		// default title "settings" (RT #145371), don't touch special pages
 		if ( $ns == NS_FORUM ) {
-			$this->title = $wgTitle->getText();
+			$this->title = $this->wg->Title->getText();
 			$this->displaytitle = false;
 			// we don't want htmlspecialchars for SpecialPages (BugId:6012)
 		} else if ( $ns == NS_SPECIAL ) {
 			$this->displaytitle = true;
 		} else if ( $ns != NS_SPECIAL ) {
 			$this->displaytitle = true;
-			$this->title = $wgOut->getPageTitle();
+			$this->title = $this->wg->Out->getPageTitle();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)